jetbrains-core/src/software/aws/toolkits/jetbrains/services/iam/Iam.kt

```diff
@@ -7,4 +7,16 @@ import software.amazon.awssdk.services.iam.IamClient
 import software.amazon.awssdk.services.iam.model.Role
 import kotlin.streams.asSequence
 
-fun IamClient.listRolesFilter(predicate: (Role) -> Boolean): Sequence<Role> = this.listRolesPaginator().roles().stream().asSequence().filter(predicate)
\ No newline at end of file
+fun IamClient.listRolesFilter(predicate: (Role) -> Boolean): Sequence<Role> = this.listRolesPaginator().roles().stream().asSequence().filter(predicate)
+
+data class IamRole(val name: String, val arn: String) {
+    override fun toString(): String = name
+
+    companion object {
+        private val ARN_REGEX = Regex("arn:.+:iam::.+:role/(.+)")
+        fun fromArn(arn: String): IamRole {
+            val name = ARN_REGEX.matchEntire(arn)?.groups?.elementAtOrNull(1)?.value ?: arn
+            return IamRole(name, arn)
+        }
```

Comment:
If the `arn` is invalid, should we throw exception instead of returning an invalid `Role`?